### PR TITLE
source/pci: add unit test for the pci source

### PR DESCRIPTION
**Pull request by marquiz**
Test PCI feature discovery against a mocked sysfs.

### CODE REVIEW

File: source/pci/pci_test.go [hunk 1]
```diff
@@ -32,4 +44,108 @@ func TestPciSource(t *testing.T) {
...
+		Instances: map[string]nfdv1alpha1.InstanceFeatureSet{
+			"device": nfdv1alpha1.InstanceFeatureSet{
+				Elements: []nfdv1alpha1.InstanceFeature{
+					{Attributes: map[string]string{
+						"class":            "0880",
+						"device":           "2021",
+						"subsystem_device": "35cf",
+						"subsystem_vendor": "8086",
+						"vendor":           "8086",
+					}},
```
[the remark this answers is not in the document]
Comment (marquiz) [reply source/pci/pci_test.go]: Yeah, better this way. Changed as suggested

File: source/pci/pci_test.go [hunk 2]
```diff
@@ -32,4 +44,108 @@ func TestPciSource(t *testing.T) {
 	assert.Nil(t, err, err)
 	assert.Empty(t, l)
 
+	// Test against our mock sysfs
+	hostpath.SysfsDir = hostpath.HostDir(filepath.Join(packagePath, "testdata", "rootfs-1", "sys"))
+
+	// Discover mock PCI devices
+	src.Discover()
```
[the remark this answers is not in the document]
Comment (marquiz) [reply source/pci/pci_test.go]: Good point! Fixed

**Review by AhmedThresh — APPROVED** [review]
LGTM, thanks @marquiz!

**Comment by marquiz** [comment]
/retitle source/pci: add unit test for the pci source

**Comment by marquiz** [comment]
I might improve this a bit
/hold

**Comment by marquiz** [comment]
Updated with more test cases
/unhold

**Pull request merged**